test/test_utils.py

```diff
@@ -369,6 +371,69 @@ def test_is_in_or_equal():
     assert not is_in_or_equal("/safe_dir/subdir/../../unsafe_file.txt", "/safe_dir/")
 
 
+def create_path_string():
```

Comment:
very interesting

Reply:
Is this deterministic or can we make it so with a seed? Otherwise it'll lead to flaky tests

Reply:
I think we can seed in CI but I think the benefit of the random fuzzing is that it can eventually find a case we have overlooked so I won't seed when running locally

Reply:
👍 and if we ever find a failure, we can add it to the known edge cases below